pkg/ds/daemon_set.go

```diff
@@ -315,12 +329,21 @@ func (ds *daemonSet) removePods() error {
 	toUnscheduleSorted := types.NewNodeSet(currentNodes...).Difference(types.NewNodeSet(eligible...)).ListNodes()
 	ds.logger.NoFields().Infof("Need to unschedule %d nodes", len(toUnscheduleSorted))
 
+	if len(toUnscheduleSorted) > 0 {
+		ds.CancelReplication()
+	}
+
 	for _, node := range toUnscheduleSorted {
 		err := ds.unschedule(node)
 		if err != nil {
 			return util.Errorf("Error unscheduling node: %v", err)
 		}
 	}
+
+	if len(podLocations)-len(toUnscheduleSorted) > 0 {
+		return ds.PublishToReplication()
```

Comment:
When I cancel the replication a few lines above, everything will be canceled and we won't know what got scheduled and what didn't by replication.  It is possible that we only need to unschedule one pod but then `ds.CancelReplication()` will halt all scheduling.
So it's possible that not all pods have been scheduled and we will need to reschedule this.